randrctl/ctl.py

```diff
@@ -193,29 +194,80 @@ def _config_safe(self, config: dict, section: str, property: str):
 
     def _configs(self):
         """
-        :return: generator of config files in defined home directories if exist
+        :return: dictionary of config files in defined home directories if exist
         """
+        # Empty container to hold the configs
+        configs = {}
         for homedir in self.homes:
-            config = os.path.join(homedir, self.config_name)
-            if os.path.isfile(config):
-                yield config
+            # Configs will be either user's (prefered) or system-wide (other)
+            is_favourite = 'preferred' if homedir == self.preferred_home else 'other'
+            # Create new entry
+            configs[is_favourite] = []
+            # Load both YAML and INI config files
+            for config_format in self.config_name:
+              config = os.path.join(homedir, config_format)
+              if os.path.isfile(config):
+                  configs[is_favourite].append(config)
+
+        # Return dictionary with the configuration files
+        return configs
+
+    def _load_config_files(self):
+        """
+        :return: dictionary of configurations loaded from config files
+        """
+        # Get config file paths
+        config_files = self._configs()
 
-    def get_randrctl(self):
-        config_files = list(self._configs())
-        config_files.reverse()  # reverse, so config file from the preferred home comes last (has highest precedence)
+        # If available, use preferred configs, otherwise use the others
+        use = 'preferred' if config_files.get('preferred') else 'other'
+
+        if use == 'other':
+          logger.warning("using system-wide configuration files")
```

Comment:
I wouldn't log this as warning. How about just logging as debug the name of the config that was picked eventually?